pkg/tfbridge/provider.go

```diff
@@ -182,6 +183,45 @@ func (e CheckFailureError) Error() string {
 	return fmt.Sprintf("CheckFailureErrors %s", e.Failures)
 }
 
+// callWithRecover is a generic function that takes a resource URN, a recovery
+// function, and a callable function as parameters. It attempts to call the
+// provided callable function and returns its results. If the callable function
+// panics, callWithRecover will recover from the panic and call the recovery
+// function with the resource URN and the panic value. The recovery function is
+// expected to convert the panic value into an error, which is then returned by
+// callWithRecover. This function is useful for handling panics in a controlled
+// manner and converting them into errors.
+func callWithRecover[T any](
+	urn resource.URN,
+	rec func(resource.URN, any) error,
+	call func() (T, error),
+) (_ T, err error) {
```

Comment:
Nit: In general, we should avoid doc comments that explain the function signature or the inner workings of the function body. Personally, I prefer a short "x does y" line, then any requirements (beyond their type) for arguments and basic usage of the function/type.

```suggestion
// callWithRecover calls f and returns it's result.
//
// If f panics, then the panic is caught and given to rec, which
// converts the panic payload into an error. rec *must* return a 
// non-nil error.
//
// callWithRecover is used for calling into foreign code and handling
// downstream panics in a controlled manor.
func callWithRecover[T any](
	urn resource.URN,
	rec func(resource.URN, any) error,
	f func() (T, error),
) (_ T, err error) {
```